torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1299,12 +1299,16 @@ def _materialize_module(
         ``reset_parameters()``, and for torchdistX fake tensors, this calls
         ``deferred_init.materialize_module()``.
         """
-        is_meta_module = any(p.is_meta for p in module.parameters())
-        is_torchdistX_deferred_init = (
-            not is_meta_module
-            and _TORCHDISTX_AVAIL
-            and any(fake.is_fake(p) for p in module.parameters())
-        )
+        try:
+            is_meta_module = any(p.is_meta for p in module.parameters())
+            is_torchdistX_deferred_init = (
+                not is_meta_module
+                and _TORCHDISTX_AVAIL
+                and any(fake.is_fake(p) for p in module.parameters())
+            )
+        except RuntimeError:
```

Comment:
This is not good. `ShardedTensor` does not support `is_meta`, so we get an error in `__torch_function__()`:
https://github.com/pytorch/pytorch/blob/270e5e519d98868af0166f3a179b286682cfb267/torch/distributed/_shard/sharded_tensor/api.py#L1085-L1087